src/canvas.py

```diff
@@ -142,14 +144,25 @@ def _set_mouse_fraction(self, event):
             self._xfrac, self._yfrac = None, None
 
     def _on_zoom_gesture(self, _gesture, scale):
-        scale = 1 + 0.05 * (scale - 1)
+        scale = 1 + 0.02 * (scale - 1)
         if scale > 5 or scale < 0.2:
             # Don't scale if ridiculous values are registered
             return
         self.zoom(scale)
 
+    def _on_pan_gesture(self, _event_controller, x, y):
+        if self.get_application().get_ctrl() is False:
+            for ax in self.axes:
+                xmin, xmax, ymin, ymax = \
+                    self._calculate_pan_values(ax, x, y)
+                ax.set_xlim(xmin, xmax)
+                ax.set_ylim(ymin, ymax)
+            self.queue_draw()
```

Comment:
A problem with both panning and zooming now is, that it doesn't get added to the view history. A proper implementation for that can be part of another pr however.

Reply:
Both gestures seem to emit end signals, so this should be easy to implement. Will take a look.